src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToSearchResultProvider.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Navigation;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.VisualStudio.Language.NavigateTo.Interfaces;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.NavigateTo
+{
+    [Export(typeof(INavigateToSearchResultProvider)), Shared]
```

Comment:
not probably in this changeset, but in design meeting, probably want to discuss whether these interface will use pure MEF or something like workspaceservice to tie service to specific workspace + more metadata?